src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -293,6 +412,130 @@ public GATKRead next()
         private static final int MIN_EVIDENCE = 5; // minimum evidence count in a cluster
     }
 
+    // Class that acts as a mapper from a stream of reads to a stream of <Kmer,BreakpointId> pairs.
+    private static final class ClusterKmerizer implements Iterator<Tuple2<Kmer,Long>>, Iterable<Tuple2<Kmer,Long>>
+    {
+        private static final int CLUSTER_BREAK_DISTANCE = 500;
+
+        public ClusterKmerizer( Integer partIdx, Iterator<GATKRead> someInputIterator, Set<Kmer> someKillSet )
+        {
+            inputIterator = someInputIterator;
+            killSet = someKillSet;
+            breakpointId = partIdx.longValue() << 32;
```

Comment:
Actually I think I'm still a little confused. Won't this assign all the kmers that are part of suspicious reads, and in the same partition on the same contig, to a single breakpoint ID? Don't we want to keep separate breakpoint ID's for each cluster?